### ISSUE TITLE
Segfault on launch with newworld ROM

### ISSUE THREAD

**Issue by rakslice** [kanjitalk755/macemu#46]
Since 5b31736785586493e3a98c2eca91f1e548765dbe there is an invalid read of 0x50f14000 on launch with newworld ROMs (I've tried 1.4 and 1.6).

This is on Windows and Linux, JIT and no-JIT, with and without the bincue feature enabled, etc.

Reverting the change to `rsrc_patches.cpp` in 5b31736785586493e3a98c2eca91f1e548765dbe makes the problem go away.

```
$ ./SheepShaver.5b31736785586493e3a98c2eca91f1e548765dbe.exe
SIGSEGV
  pc 0109ae00
  ea 61f14000
 r0 00000000   r1 17ff5890   r2 00000000   r3 50f14000
 r4 00000040   r5 fffffffa   r6 00003540   r7 ffffffe8
 r8 00110477   r9 61776163  r10 0000bf8c  r11 00000384
r12 0000001e  r13 61776163  r14 00000000  r15 00000001
r16 001e0f4e  r17 000a7ef0  r18 0011fdc0  r19 0010d05a
r20 17ff5a9e  r21 001a68b6  r22 17ff59da  r23 00000000
r24 001e0f7e  r25 00000000  r26 00000000  r27 000050f1
r28 001be0c4  r29 40c89e00  r30 40c60000  r31 68fff000
 f0 0.00000   f1 0.00000   f2 0.00000   f3 0.00000
 f4 0.00000   f5 0.00000   f6 0.00000   f7 0.00000
 f8 0.00000   f9 0.00000  f10 0.00000  f11 0.00000
f12 0.00000  f13 0.00000  f14 0.00000  f15 0.00000
f16 0.00000  f17 0.00000  f18 0.00000  f19 0.00000
f20 0.00000  f21 0.00000  f22 0.00000  f23 0.00000
f24 0.00000  f25 0.00000  f26 0.00000  f27 0.00000
f28 0.00000  f29 0.00000  f30 0.00000  f31 0.00000
 lr 40c89e00  ctr 00000000   cr 40001c22  xer 00000000
 pc 40c6628c fpscr 00000000
SheepShaver V2.5 by Christian Bauer and Mar"c" Hellwig
Reading ROM file...
Using SDL_Renderer driver: software
Detected CPU features: MMX SSE SSE2 SSE3 SSSE3
PowerPC CPU emulator by Gwenole Beauchesne
WARNING: Unknown DiskStatus(6)
  0x40c6626c:  sth     r4,0(r3)
  0x40c66270:  addco.  r4,r4,r0
  0x40c66274:  bgelr+  cr2
  0x40c66278:  b       0x40c6d0d4
  0x40c6627c:  nop
  0x40c66280:  rlwimi  r29,r27,3,13,28
  0x40c66284:  mtlr    r29
  0x40c66288:  lhau    r27,2(r24)
 >0x40c6628c:  stb     r4,0(r3)
  0x40c66290:  addco.  r4,r4,r0
  0x40c66294:  bgelr+  cr2
  0x40c66298:  b       0x40c6d0d4
  0x40c6629c:  lhau    r27,2(r24)
  0x40c662a0:  addco.  r4,r4,r0
  0x40c662a4:  lhau    r6,2(r24)
  0x40c662a8:  stw     r4,0(r27)
```

**Comment by rakslice** [comment]
I don't know what the change to the sound input driver in that merge is for, but I presume it is not intended to break compatibility with newworld ROM?

**Comment by kanjitalk755** [comment]
@spolsley

It seems that there is no problem in the operation of bin/cue even if rsrc_patches.cpp is reverted.
Can we revert it?

**Comment by spolsley** [comment]
Yes, it looks like it can be reverted.  I'd initially removed that patch since the soundin driver from audio.cpp wasn't loading, but I'm not seeing a difference in behavior now, so it was probably some other audio change.  It was motivated by a couple games that need to see the CD as an input.

The patch didn't change loading of any new world or old world roms I tried, but if it's breaking some compatibility, better to revert.  I am happier to put it back in since it'll be needed to extend sound input capabilities to Sheepshaver.

I can put in a PR to fix it shortly if that'll be helpful! :-)

**Comment by spolsley** [comment]
Whoops, I didn't test both -16501 and -16500.  The latter is what patches the audio.cpp sound input driver.  For me, this driver still isn't fully loading, and certain games (e.g., Prime Target) will be unable to recognize a CDROM for audio playthrough.

However, -16501 is what causes the crash if not patched.  I didn't notice because I use ignoresegv and ignoreillegal to true.

@rakslice I'll put up a PR which restores the -16501 and not -16500.  Would you be able to confirm if this runs on your configuration?  I do still have an interest in completing the sound input implementation in audio.cpp, but if we don't need to patch -16500 for operation, I think Sheepshaver will run more things in the interim than attempting to use the existing broken sound input driver.

**Comment by rakslice** [comment]
@spolsley The PR still had the problem, as I noted there.

**Comment by spolsley** [comment]
@rakslice I meant to ask if the new sound patch resolves this issue for you.  Have you encountered this crash since?  If not, I think this can be closed as of #47.

**Comment by rakslice** [comment]
Yes, builds from the current codebase where the rsrc_patch for DRVR -16501, -16500 are back eliminates this crash.
